Firestore/core/src/firebase/firestore/local/leveldb_key.cc

```diff
@@ -502,266 +469,347 @@ std::string Describe(leveldb::Slice key) {
   return description;
 }
 
+class Writer {
+ public:
+  std::string result() const {
+    return dest_;
+  }
+
+  void WriteTerminator() {
+    OrderedCode::WriteSignedNumIncreasing(&dest_, ComponentLabel::Terminator);
+  }
+
+  void WriteTableName(const char *table_name) {
+    WriteLabeledString(ComponentLabel::TableName, table_name);
+  }
+
+  void WriteBatchId(model::BatchId batch_id) {
+    WriteLabeledInt32(ComponentLabel::BatchId, batch_id);
+  }
+
+  void WriteCanonicalId(absl::string_view canonical_id) {
+    WriteLabeledString(ComponentLabel::CanonicalId, canonical_id);
+  }
+
+  void WriteTargetId(model::TargetId target_id) {
+    WriteLabeledInt32(ComponentLabel::TargetId, target_id);
+  }
+
+  void WriteUserId(absl::string_view user_id) {
+    WriteLabeledString(ComponentLabel::UserId, user_id);
+  }
+
+  /**
+   * For each segment in the given resource path writes an
```

Comment:
You're right. This comes from the fact that the previous enumeration value here was "FSTComponentLabelPathSegment". In that case "an" worked because we spell the prefix out. Now that the prefix is gone it should be "a".